storage-bigtable/src/access_token.rs

```diff
@@ -91,41 +105,55 @@ impl AccessToken {
     }
 
     /// Call this function regularly to ensure the access token does not expire
-    pub async fn refresh(&self) {
+    pub fn try_refresh(&self) {
         // Check if it's time to try a token refresh
-        {
-            let token_r = self.token.read().unwrap();
-            if token_r.1.elapsed().as_secs() < token_r.0.expires_in() as u64 / 2 {
-                return;
+        match self.token.read() {
+            Ok(token_r) => {
+                if token_r.1.elapsed().as_secs() < token_r.0.expires_in() as u64 / 2 {
+                    debug!("token is not expired yet");
+                    return;
+                }
+                drop(token_r);
             }
-
-            #[allow(deprecated)]
-            if self
-                .refresh_active
-                .compare_and_swap(false, true, Ordering::Relaxed)
-            {
-                // Refresh already pending
+            Err(_poisoned) => {
+                error!("token is poisoned (read op)");
```

Comment:
This change is just to print an error message, correct? Are you actually seeing the RwLock being poisoned ever?
If this change is needed, it should be a separate commit.